.github/documentation.md

```diff
@@ -0,0 +1,202 @@
+
+#### Dependency Image Build Pipeline Outputs
+
+* `base-spack` Docker image: Of the form `base-spack-<compiler name><compiler version>-<spack_packages version>:latest`. A docker image that contains a `spack` install, `access-nri/spack_packages` repo at the specified version, and a site-wide compiler for spack to use to build dependencies.
+* `dependency` Docker image: Of the form `build-<coupled model>-<compiler name><compiler version>-<spack_packages version>:latest`: A docker image based on the above `base-spack` image, that contains all the model components dependencies (separated by `spack env`s), but not the models themselves. The models are added on top of the install in a different pipeline, negating the need for a costly install of the dependencies again (in most cases).
+
+### I/O for Model Test Pipeline
+
+#### Model Test Pipeline Inputs
+
+There are no explicit inputs to this workflow. The information required is inferred by the model repository that calls the `build-package.yml` workflow.
+
+However, there are indirect inputs into this pipeline:
+
+* Appropriate `dependency` Docker images of the form: `ghcr.io/access-nri/build-<coupled model>-<compiler name><compiler version>-<spack_packages version>:latest`.
+* `containers/compilers.json`: This is a structure of all the compilers we want to test against.
+* `containers/models.json`: This is a structure of all the coupled models (and their associated model components) that we want to test against.
+
+#### Model Test Pipeline Outputs
+
+* `<env>.original.spack.lock`: A `spack.lock` file from the `spack env` associated with the modified model repository (eg. `mom5`), before the installation of the modified model. If the install succeeds then then this `spack.lock` file was unmodified during the installation and can be used to recreate this `spack env`. This file is uploaded as an artifact.
+* Optionally, a `<env>.force.spack.lock`: If the installation fails (namely, if installing the modified model would change the `spack.lock` file) we force a regeneration of the `spack.lock` file and upload this as an artifact as well.
+
+### I/O for JSON Linter Pipeline
+
+#### Inputs for JSON Linter Pipeline
+
+This pipeline has no explicit inputs.
+
+However, there are indirect inputs into this pipeline:
+
+* `*.json`: All JSON files that need to be validated.
+* `*.schema.json`: All JSON schemas that validate the above `*.json` files.
+
+#### Outputs for JSON Linter Pipeline
+
+There are no outputs from this pipeline (outside of checks).
+
+## CI Workflow Run Through
+
+### Dependency Image Build Pipeline
+
+The rationale for this pipeline is the creation of a model-dependency docker image. This image contains spack, the `spack env`ironments, and the dependencies for the install of a model, but not the model itself. This allows modified models/model components to be 'dropped in' to the dependency image and installed quickly, without needing to install the dependencies again.
```

Comment:
Choose between "`spack env`" and "spack environments".

Reply:
Choosing `spack env` over "spack environments" as it is also a command and can give a breadcrumb for readers who want to learn more about environments.